demo/lit-element.html

```diff
@@ -104,24 +104,31 @@ <h4 on-click="${(e) => console.log(this, e.target)}">Foo: ${foo}, Bar: ${bar}</h
         console.log('updated!', changedProps);
       }
 
-      async finishUpdate(changedProps) {
-        if (!this._inner) {
-          this._inner = this.shadowRoot.querySelector('x-inner');
+      finishFirstUpdate(changedProps) {
+        this._inner = this.shadowRoot.querySelector('x-inner');
+      }
+
+      finishUpdate() {
+        if (this.whales < 100) {
+          this.whales++;
         }
-        await this._inner.updateComplete;
       }
-    }
 
-    addEventListener('WebComponentsReady', function() { console.log('wcr - module')})
+      get updateComplete() {
+        return (async () => {
+          await super.updateComplete;
+          await this._inner.updateComplete;
+          while (!await super.updateComplete) {};
```

Comment:
`return true;` after the while loop to indicate that the `updateComplete` is actually fully complete (no new updates scheduled)

Reply:
Done and fixed other tests. Thanks!